contracts/LoyaltyCard.sol

```diff
@@ -0,0 +1,355 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.11;
+
+// Interface
+import "./interfaces/ILoyaltyCard.sol";
+
+// Base
+import "./eip/ERC721AVirtualApproveUpgradeable.sol";
+
+// Lib
+import "./lib/CurrencyTransferLib.sol";
+
+// Extensions
+import "./extension/NFTMetadata.sol";
+import "./extension/SignatureMintERC721Upgradeable.sol";
+import "./extension/ContractMetadata.sol";
+import "./extension/Ownable.sol";
+import "./extension/Royalty.sol";
+import "./extension/PrimarySale.sol";
+import "./extension/PlatformFee.sol";
+import "./extension/Multicall.sol";
+import "./extension/PermissionsEnumerable.sol";
+import "./extension/DefaultOperatorFiltererUpgradeable.sol";
+import "./openzeppelin-presets/metatx/ERC2771ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+contract LoyaltyCard is
```

Comment:
would be great to have some doc here about what this contract is and what the main features are